test/unit/stats_test.lua

```diff
@@ -0,0 +1,556 @@
+
+                return state + 1, param
+            end
+            rawset(_G, 'sleep_ten_times', sleep_ten_times)
+        ]],
+        eval = [[
+            local params, space_name, op = ...
+            local sleep_time = params.sleep_time
+
+            local build_sleep_multiplier = 2
+
+            local wrapped = stats_module.wrap(
+                function(space_name)
+                    sleep_for(build_sleep_multiplier * sleep_time)
+
+                    return sleep_ten_times, {}, 0
+                end,
+                op,
+                { pairs = true }
+            )
+
+            for i, _ in wrapped(space_name) do
+                if i == 5 then
+                    break
+                end
+            end
+        ]],
+        post_eval = [[
+            collectgarbage('collect')
+            collectgarbage('collect')
+            require('fiber').yield()
+        ]],
+        build_sleep_multiplier = 2,
+        iterations_expected = 5,
+        changed_coll = 'ok',
+        unchanged_coll = 'error',
+    }
+}
+
+for name, case in pairs(pairs_cases) do
+    local test_name = ('test_pairs_wrapper_observes_all_iterations_on_%s'):format(name)
+
+    g.before_test(test_name, function(g)
+        g.router:eval(case.prepare, { helpers.simple_functions_params() })
+    end)
+
+    g[test_name] = function(g)
+        local op = stats_module.op.SELECT
+
+        local params = helpers.simple_functions_params()
+        local args = { params, space_name, op }
+
+        local before_start = clock.monotonic()
+
+        if case.pcall then
+            pcall(g.router.eval, g.router, case.eval, args)
+        else
+            g.router:eval(case.eval, args)
+        end
+
+        if case.post_eval then
+            g.router:eval(case.post_eval)
+        end
+
+        local after_finish = clock.monotonic()
+        local time_diff = after_finish - before_start
+
+        -- Validate stats format after execution.
+        local total_stats = g:get_stats()
+        t.assert_type(total_stats, 'table', 'Total stats present after observations')
+
+        local space_stats = g:get_stats(space_name)
+        t.assert_type(space_stats, 'table', 'Space stats present after observations')
+
+        t.assert_equals(total_stats.spaces[space_name], space_stats,
+            'Space stats is a section of total stats')
+
+        local op_stats = space_stats[op]
+        t.assert_type(op_stats, 'table', 'Op stats present after observations for the space')
+
+        -- Expected collectors (changed_coll: 'ok' or 'error') have changed.
+        local changed = op_stats[case.changed_coll]
+        t.assert_type(changed, 'table', 'Status stats present after observations')
+
+        t.assert_equals(changed.count, 1, 'Count incremented by 1')
+
+        t.assert_ge(changed.latency,
+            params.sleep_time * (case.build_sleep_multiplier + case.iterations_expected),
+            'Latency has appropriate value')
```

Comment:
Not sure I got the idea, but should not we multiply all three values?

Reply:
We build initial `gen, param, state` for `params.sleep_time * case.build_sleep_multiplier` and then do `case.iterations_expected` with `params.sleep_time` each.

Reply:
Ah, **build**_sleep_multiplier. Got it.